src/libecalc/core/consumers/base/component.py

```diff
@@ -22,15 +20,8 @@ def evaluate(
 class BaseConsumerWithoutOperationalSettings(ABC):
     id: str
 
-    @property
     @abstractmethod
-    def operational_settings(self):
-        ...
-
-    @abstractmethod
-    def get_max_rate(
-        self, operational_settings: Union[CompressorOperationalSettings, PumpOperationalSettings]
-    ) -> List[float]:
+    def get_max_rate(self, inlet_stream: Stream, target_pressure: TimeSeriesFloat) -> List[float]:
```

Comment:
this one also gets more complicated when/if we allow multiple streams in/out of a consumer/compressor train/system/network. 

`def get_max_rate_per_stream(self, streams: List[Stream]) -> List[float]`

or something. The maximum of one stream given that all others are kept constant?